src/utils/dom/selectNodesWithShadow.js

```diff
@@ -54,8 +54,15 @@ export default (context, selector) => {
   // find each subselector element based on the previously selected node's shadowRoot
   let parent = context;
   for (let i = 0; i < parts.length; i += 1) {
```

Comment:
could you please refactor to use `forEach` loop?

Reply:
Unfortunately I can't refactor to a `forEach` since I need to use `continue` and also do an early return, and it would look much less readable if I refactor it to a `forEach` with an external temp variable.